### DISCUSSION TITLE
Building, installing and linking with six-library

### DISCUSSION THREAD

**Discussion by vmartinlac** [ngageoint/six-library#720] [Q&A]
Hello,

How to build, install and link against six-library?

If I pull master, I build with default options (with CMake and ninja on Ubuntu 22.04). I install. I write some CMake listing to detect six-library and it tells me it cannot find package "coda-oss". So I install coda-oss. Then it tells me that it cannot find CodaFindSystemDependencies.cmake (it does not get copied when installing).

If I use release 3.2.4, compilation fails with message "/home/....../archives/six-library-SIX-3.2.4/externals/coda-oss/modules/c++/except/include/except/Throwable.h:160:25: error: Declaring method ‘virtual std::string except::Throwable::toString() const’ final would enable devirtualization of 1 call [-Werror=suggest-final-methods]
  160 |     virtual std::string toString() const"

Best regards

**Comment by JDanielSmith** [comment]
```
e04) git checkout main
Already on 'main'
Your branch is up to date with 'origin/main'.
e04) rm -r -f out && mkdir out && cd out
e04) cmake ..
[...]
e04) cmake --build . -j
[...]
e04) ctest
[...]
100% tests passed, 0 tests failed out of 138

Total Test time (real) =  49.79 sec
```

**Reply by eleehiga** [reply]
Could you guys please add this to the instructions

**Comment by vmartinlac** [comment]
Hello,

The following bash script

```
ROOT=$(pwd)
git clone https://github.com/ngageoint/six-library.git
mkdir build
cd build
cmake ../six-library -G Ninja -DCMAKE_INSTALL_PREFIX=${ROOT}/install -DCMAKE_BUILD_TYPE=Release
ninja install
cd ..
mkdir test
cd test
cat > CMakeLists.txt << EOM
cmake_minimum_required(VERSION 3.0)
project(tmp)
find_package(six-library REQUIRED)
EOM
cmake . -Dsix-library_DIR="${ROOT}/install/lib/cmake"
```
gives
```
CMake Error at /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "Findcoda-oss.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "coda-oss",
  but CMake did not find one.

  Could not find a package configuration file provided by "coda-oss" with any
  of the following names:

    coda-ossConfig.cmake
    coda-oss-config.cmake

  Add the installation prefix of "coda-oss" to CMAKE_PREFIX_PATH or set
  "coda-oss_DIR" to a directory containing one of the above files.  If
  "coda-oss" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /home/victor/archives/six/install/lib/cmake/six-libraryConfig.cmake:29 (find_dependency)
  CMakeLists.txt:3 (find_package)
```
If I manually install coda-oss, cmake find_package tells that CodaFindSystemDependencies.cmake is missing.

**Comment by vmartinlac** [comment]
What is the recommended way to configure linking against six-library? The typical cmake way install / find_package seems not to work.
